dialogue-clients/src/main/java/com/palantir/dialogue/clients/DialogueDnsResolutionWorker.java

```diff
@@ -103,6 +109,8 @@ private synchronized void doUpdate(@Nullable INPUT updatedInputState) {
             } else {
                 log.info("Attempted to update DNS output refreshable which has already been garbage collected");
             }
+            long end = System.nanoTime();
+            updateTimer.update(Duration.ofNanos(end - start));
```

Comment:
nit: let's avoid Duration allocation and use `updateTimer.update(end - start, TimeUnit.NANOSECONDS)`

Reply:
fixed